src/reorderable/ReorderableListItemComponent.coffee

```diff
@@ -0,0 +1,104 @@
+_ = require 'lodash'
+React = require 'react'
+H = React.DOM
+R = React.createElement
+ReactDOM = require 'react-dom'
+
+DragSource = require('react-dnd').DragSource
+DropTarget = require('react-dnd').DropTarget
+
+itemTarget =
+  drop: (props, monitor, component) ->
+    dragItemIndex = monitor.getItem().index
+    hoverItemIndex = props.index
+
+    props.updateOrder(dragItemIndex, hoverItemIndex)
+    return {}
+
+  hover: (props, monitor, component) ->
+    dragIndex = monitor.getItem().index
+    hoverIndex = props.index
+
+    if dragIndex == hoverIndex
+      return
+
+    if props.constrainTo != monitor.getItem().constrainTo
+      return
+
+    hoverBoundingRect = ReactDOM.findDOMNode(component).getBoundingClientRect()
```

Comment:
All this needs at least a few comments for the next person who has to look at it. Not crazy amounts, but enough to easily make sense of it.